libraries/hotstuff/include/eosio/hotstuff/qc_chain.hpp

```diff
@@ -30,6 +33,34 @@ namespace eosio::hotstuff {
    using namespace boost::multi_index;
    using namespace eosio::chain;
 
+   template <typename StateObj>
+   static void read_state(const std::string file_path, StateObj& s){
+
+      if (file_path != std::string()){
+         fc::cfile pfile;
+         pfile.set_file_path(file_path);
+         pfile.open("rb");
+         auto ds = pfile.create_datastream();
+         fc::raw::unpack(ds, s);
+         pfile.close();
```

Comment:
I don't think we want to `open`/`close` the file for each read.

Reply:
In the current code, the safety state file is read only once, so I'm doing open/close in that case. The added state_db_manager (in the latest commit) however has support to do multiple reads with the same open file handle if that's ever needed.